mllib/src/main/scala/org/apache/spark/mllib/feature/PCA.scala

```diff
@@ -44,6 +44,13 @@ class PCA @Since("1.4.0") (@Since("1.4.0") val k: Int) {
     require(k <= numFeatures,
       s"source vector size $numFeatures must be no less than k=$k")
 
+    val workSize = ( 3
```

Comment:
I know it's copied from Breeze, but this is pretty unreadable and duplicates code. It's a bit fragile too. This doesn't really help, it just causes a different exception to be thrown.

Reply:
Hmm... The intention is provide some user-friendly exception information because our customer meet this problem. Breeze will throw `NegativeArraySizeException` which user cannot understand. But we can check and throw more detail info to tell user to decrease the `k` and `numFeatures` to avoid this issue.

Reply:
Definitely, the error message is much better.  How about adding a Github permalink to the Breeze code this is from, for future reference?

Reply:
OK, how about catching NegativeArraySizeException and rethrowing a better error? replicating the check is really what I'm questioning, as it's pretty obtuse.

Reply:
But catching `NegativeArraySizeException` won't be very precise, other place also possible to throw `NegativeArraySizeException`. It is a very common exception.